pythclient/price_feeds.py

```diff
@@ -407,3 +410,118 @@ def price_attestation_to_price_feed(price_attestation):
+def extract_price_info_from_accumulator_update(
+    price_feed_id, update_data, encoding
+) -> Optional[Dict[str, Any]]:
+    offset = 0
+    offset += 4  # magic
+    offset += 1  # major version
+    offset += 1  # minor version
+
+    trailing_header_size = update_data[offset]
+    offset += 1 + trailing_header_size
+
+    update_type = update_data[offset]
+    offset += 1
+
+    if update_type != 0:
+        print(f"Invalid accumulator update type: {update_type}")
+        return None
+
+    vaa_length = int.from_bytes(update_data[offset : offset + 2], byteorder="big")
+    offset += 2
+
+    vaa_buffer = update_data[offset : offset + vaa_length]
+    # convert vaa_buffer to string based on encoding
+    if encoding == "hex":
+        vaa_str = vaa_buffer.hex()
+    elif encoding == "base64":
+        vaa_str = base64.b64encode(vaa_buffer).decode("ascii")
+    parsed_vaa = parse_vaa(vaa_str, encoding)
+    offset += vaa_length
+
+    num_updates = update_data[offset]
+    offset += 1
+
+    for _ in range(num_updates):
+        message_length = int.from_bytes(
+            update_data[offset : offset + 2], byteorder="big"
+        )
+        offset += 2
+
+        message = update_data[offset : offset + message_length]
+        offset += message_length
+
+        proof_length = update_data[offset]
+        offset += 1
+        offset += proof_length  # ignore proofs
+
+        message_offset = 0
+        message_type = message[message_offset]
+        message_offset += 1
+
+        if message_type != 0:
+            continue
+
+        price_id = message[message_offset : message_offset + 32].hex()
+        message_offset += 32
+
+        if price_id != price_feed_id:
+            continue
+
+        price = int.from_bytes(
+            message[message_offset : message_offset + 8], byteorder="big", signed=True
+        )
+        message_offset += 8
+        conf = int.from_bytes(
+            message[message_offset : message_offset + 8], byteorder="big", signed=False
+        )
+        message_offset += 8
+        expo = int.from_bytes(
+            message[message_offset : message_offset + 4], byteorder="big", signed=True
+        )
+        message_offset += 4
+        publish_time = int.from_bytes(
+            message[message_offset : message_offset + 8], byteorder="big", signed=True
+        )
+        message_offset += 8
+        prev_publish_time = int.from_bytes(
+            message[message_offset : message_offset + 8], byteorder="big", signed=True
+        )
+        message_offset += 8
+        ema_price = int.from_bytes(
+            message[message_offset : message_offset + 8], byteorder="big", signed=True
+        )
+        message_offset += 8
+        ema_conf = int.from_bytes(
+            message[message_offset : message_offset + 8], byteorder="big", signed=False
+        )
+
+        return PriceInfo(
+            seq_num=parsed_vaa["sequence"],
+            vaa=vaa_str,
```

Comment:
here we need to pass the whole updateData not the actual vaa inside it. (in general this is update data that is used to update the price on-chain)